hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieWriteHandle.java

```diff
@@ -336,4 +357,52 @@ protected static Option<IndexedRecord> toAvroRecord(HoodieRecord record, Schema
       return Option.empty();
     }
   }
+
+  /**
+   * Event time watermarker is tracked when
+   * 1. Record type is AVRO.
+   * 2. Event time ordering merge mode is used.
+   * 3. The flag is set.
+   */
+  boolean shouldTrackEventTimeWaterMarker(HoodieTableMetaClient metaClient,
+                                          HoodieWriteConfig config) {
+    return config.getRecordMerger().getRecordType() == AVRO
```

Comment:
why we limit the record type as avro, we have `HoodieRecord.getColumnValueAsJava` to fetch the value directly which is more efficient.